### DISCUSSION TITLE
A Proposal for Moving Forward

### DISCUSSION THREAD

**Discussion by djaqua** [node-saml/xml-crypto#305] [Ideas]
I would like to take a more pro-active approach to the next major version (4?) (and its inevitable plethora of breaking changes).

While I'm not opposed to breaking changes, I do think it can be a bit rude to frequently and continuously change the client-facing side of this project.

As high level goals, I propose the following:

An emphasis on providing a stable interface for clients
An emphasis on providing a simple & intuitive interface for clients
A rigorous overhaul of the documentation - I volunteer myself for this since I actually enjoy technical writing (I'm a sick unit)
Deprecate features that have proven to be a consistent pain point
Add crowd-pleasing functionality
I'm open to suggestions, here, but could it be worth experimenting with the "new" projects tab to manage chores/tickets?

Lets have a spectacular version 4 release!

**Comment by cjbarth** [comment]
@djaqua , I'm sorry about the radio silence on this. I've tried to make smart changes for the 4.x release and I'm actively making other minor changes in the 4.x series to improve typing and add simple features. Please feel free to contribute any PR, particularly around documentation, that you feel would move forward toward these goals. I like what you're suggesting here.
